spec/player_ai_minimax_spec.rb

```diff
@@ -0,0 +1,108 @@
+require 'player_ai_minimax'
+require 'game_board'
+require 'game_state'
+require 'game_rules'
+
+describe PlayerAiMinimax do
+  let (:io) { double }
+  let (:board) { GameBoard.new }
+  let (:rules) { GameRules.new }
+  let (:test_ai) { PlayerAiMinimax.new(:X, rules) }
+  let (:test_ai2) { PlayerAiMinimax.new(:O, rules) }
+  let (:test_state) { GameState.new(board , :X, :O) }
+  let(:spaces) { test_state.game_board.spaces }
+  let(:corners) { [0, 2, 6, 8] }
+
+  describe "It has a mark " do
+    it ".mark returns :X " do
+      expect(test_ai.mark).to eq(:X)
+    end
+  end
+  
+  describe "Playing the last move " do
+    it "plays 0 when that is the only space open " do
+      [1,2,3,4,5,6,7,8].each do |i|
```

Comment:
FYI: In ruby you could do something like (not 100% sure on the syntax but): `(1..8).each` or `(1...8)` for inclusive. Lookup ruby ranges.